File: smallestMissingNonNegative.go
package main

func findSmallestInteger(nums []int, value int) int {
    freq := make(map[int]int)
    for _, num := range nums {
        rem := ((num % value) + value) % value
        freq[rem]++
    }

    mex := 0
    for {
        rem := mex % value
        if freq[rem] > 0 {
            freq[rem]--
            mex++
        } else {
            return mex
        }
    }
}

File: smallestMissingNonNegative.py
from collections import Counter
from typing import List

class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        freq = Counter(((num % value) + value) % value for num in nums)
        mex = 0
        while True:
            rem = mex % value
            if freq[rem] > 0:
                freq[rem] -= 1
                mex += 1
            else:
                return mex
